lib/hash.js

```diff
@@ -0,0 +1,74 @@
+const crypto = require('crypto');
+
+const SALT_LENGTH = 64;
+const HASH_ITERATIONS = 10000;
+const HASH_LENGTH = 256;
+const CIPHER = 'sha512';
+const SEPARATOR = ':';
+
+function genSalt() {
+  return crypto.randomBytes(SALT_LENGTH).toString('hex');
```

Comment:
I would split it into 2 lines. generateRandomBytes and then convert it to hexadecimal for better understanding